### ISSUE TITLE
Setting for a rule enables it implicitly

### ISSUE THREAD

**Issue by kytta** [charliermarsh/ruff#3683]
I have a big Python project that I am migrating to use Ruff. For now, I disable some of the rules in order to enable them later (I do not want to break my CI just yet). However, I already defined some settings for the rules I will use in the future.

Here's a very minimal example:

```toml
# pyproject.toml

[tool.ruff]
extend-select = [
    # TODO: fix docs in a separate issue
    # "D",  # pydocstyle
    "PT",  # flake8-pytest-style
]

fixable = [
    "PT001",  # pytest fixture with parens
    "PT006",  # tuple in pytest parametrize
]

[tool.ruff.pydocstyle]
convention = "pep257"
```

```py
# main.py

def my_func(arg):
   '''Return something
   param docs are not provided'''


   return arg+1
```

## What I expect

I expect that when I run Ruff, it will only care about the enabled rules (the default `E` and `F` as well as the `PT`). `D` is not selected, so Ruff is not bothering to fix it.

## Actual behaviour
Ruff detects (and attempts to fix) the `D` issues:

```bash
$ ruff . --format=grouped --show-fixes   
warning: `multi-line-summary-first-line` (D212) and `multi-line-summary-second-line` (D213) are incompatible. Ignoring `multi-line-summary-second-line`.
myproject/main.py:
  2:4  D415 [*] First line should end with a period, question mark, or exclamation point

Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

## Workaround?

I don't know if it's expected behaviour or not. In any case, if I also comment the `[tool.ruff.pydocstyle]` out, the ruleset gets ignored properly

---

Ruff version is 0.0.258

**Comment by JonathanPlasse** [comment]
`convention` is used to select which pydocstyle use and enable the corresponding `D` rules.
https://beta.ruff.rs/docs/settings/#convention
Why did you add it to your configuration?

**Comment by kytta** [comment]
> Why did you add it to your configuration?

I generally want to enforce it, but not at this point in time, since the repo is too "dirty". That's why I temporarily disabled the rule inside `select`.

> [...] and enable the corresponding D rules.

I see, I guess it's desired behaviour, then. Thanks!

**Comment by charliermarsh** [comment]
Sorry about that, this is a confirmed regression in the latest release. You can downgrade to 0.0.257 for now, but we'll ship a fix soon.

**Comment by charliermarsh** [comment]
Should be fixed by https://github.com/charliermarsh/ruff/pull/3685, which'll go out later today.
